connector/google/google.go

```diff
@@ -227,8 +235,11 @@ func (c *googleConnector) createIdentity(ctx context.Context, identity connector
 
 	var groups []string
 	if s.Groups && c.adminSrv != nil {
-		checkedGroups := make(map[string]struct{})
-		groups, err = c.getGroups(claims.Email, c.fetchTransitiveGroupMembership, checkedGroups)
+		if c.fetchTransitiveGroupMembership {
+			groups, err = c.getAllGroups(ctx, claims.Email)
+		} else {
+			groups, err = c.getGroups(ctx, claims.Email, &sync.Map{})
```

Comment:
sync.Map is only suitable for special cases. I'd like to suggest avoiding it here.

Reply:
Fixed